node.gypi

```diff
@@ -376,8 +376,8 @@
               'conditions': [
                 ['OS in "linux freebsd" and node_shared=="false"', {
                   'ldflags': [
-                    '-Wl,--whole-archive,'
-                      '<(obj_dir)/deps/openssl/<(openssl_product)',
+                    '-Wl,--whole-archive',
+                    '<(obj_dir)/deps/openssl/<(openssl_product)',
```

Comment:
This change doesn't look correct to me. libopenssl.a is no longer passed to the linker because it's not part of -Wl any longer. Probably works with binutils/llvm by accident but I'd rather avoid relying on such toolchain quirks.

Does it work with `zig cc` when written like this?
```suggestion
                    '-Wl,--whole-archive',
                    '-Wl,<(obj_dir)/deps/openssl/<(openssl_product)',
```

Reply:
Hi,

thank you for your quick reply!

The change you suggested will result in the same linker error since it is equivalent to the orginal version. So unfortunately it did not work.

Please also have a quick look at the following quote from the GCC manpage for the "-Wl," option:
> Pass option as an option to the linker. If option contains commas, it is split into multiple options at the commas. You can use this syntax to pass an argument to the option. For example, -Wl,-Map,output.map passes -Map output.map to the linker. When using the GNU linker, you can also get the same effect with -Wl,-Map=output.map. 

As a result, the `openssl.a` file path will be passed as an option to the linker which will fail since this is not a valid linker option.

In my change I just adjusted the line to match the way how it is done for other `-Wl,--whole-archive`, `-Wl,--no-whole-archive` occurences in `node.gpyi`, e.g. for `zlib`:
```
161                 '-Wl,--whole-archive',
162                 '<(obj_dir)/deps/zlib/<(STATIC_LIB_PREFIX)zlib<(STATIC_LIB_SUFFIX)',
163                 '-Wl,--no-whole-archive',
```

or `libuv`:
```
200                 '-Wl,--whole-archive',
201                 '<(obj_dir)/deps/uv/<(STATIC_LIB_PREFIX)uv<(STATIC_LIB_SUFFIX)',
202                 '-Wl,--no-whole-archive',
```

If you check the `--whole-archive` documentation from the `ld` manpages:
> For each archive mentioned on the command line after the
> --whole-archive option, include every object file in the
> archive in the link, rather than searching the archive for
> the required object files.  This is normally used to turn an
> archive file into a shared library, forcing every object to
> be included in the resulting shared library.  This option may
> be used more than once.

to me it sounds like the way how the flags are used for the `zlib` and `libuv` dependencies is the right way to do it.
 
Note that also other people [here](https://stackoverflow.com/a/842770) or [there](https://stackoverflow.com/a/4767951) are using those flags like that.

best regards,
Fabian

Reply:
Ok I just reread the GCC manpages example:
> For example, -Wl,-Map,output.map passes
>            -Map output.map to the linker.  When using the GNU linker, you can also get the same effect with -Wl,-Map=output.map.

So this syntax is also ok!